Shared/Samples/Show geodesic path between two points/ShowGeodesicPathBetweenTwoPointsView.swift

```diff
@@ -0,0 +1,129 @@
+// Copyright 2025 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct ShowGeodesicPathBetweenTwoPointsView: View {
+    /// The map that will be displayed in the map view.
+    @State private var map = Map(basemapStyle: .arcGISImageryStandard)
+    
+    /// The graphics overlay that will be displayed on the map view.
+    /// This will hold the graphics that show the start point, end point,
+    /// and geodesic path.
+    @State private var overlay = GraphicsOverlay()
+    
+    /// The current measurement state.
+    @State private var state: MeasurementState = .notStarted {
+        didSet {
+            updateGraphicsOverlay()
+        }
+    }
+    
+    /// The symbology for point graphics.
+    private let pointSymbol: Symbol = SimpleMarkerSymbol(style: .cross, color: .blue, size: 20)
+    
+    /// The symbology for the line graphic.
+    private let lineSymbol: Symbol = SimpleLineSymbol(style: .dash, color: .yellow, width: 2)
+    
+    var body: some View {
+        MapView(map: map, graphicsOverlays: [overlay])
+            .onSingleTapGesture { _, mapPoint in
+                switch state {
+                case .notStarted, .complete:
+                    // If the state is empty or complete, then start a new
+                    // path, adding the tap point as the first graphic.
+                    state = .startOnly(start: mapPoint)
+                case .startOnly(let start):
+                    // If the state was started, then add the end point
+                    // to complete it.
+                    state = .complete(start: start, end: mapPoint)
+                }
```

Comment:
You could use a switch expression here.

```suggestion
                state = switch state {
                case .notStarted, .complete:
                    // If the state is empty or complete, then start a new
                    // path, adding the tap point as the first graphic.
                    .startOnly(start: mapPoint)
                case .startOnly(let start):
                    // If the state was started, then add the end point
                    // to complete it.
                    .complete(start: start, end: mapPoint)
                }
```